src/main/java/net/rubyeye/xmemcached/impl/AddressMemcachedSessionComparator.java

```diff
@@ -0,0 +1,29 @@
+package net.rubyeye.xmemcached.impl;
+
+import java.io.Serializable;
+import net.rubyeye.xmemcached.MemcachedSessionComparator;
+import net.rubyeye.xmemcached.networking.MemcachedSession;
+import com.google.code.yanf4j.core.Session;
+
+/**
+ * Connection comparator,compare with Address
+ * 
+ * @author Jungsub Shin
+ * 
+ */
+public class AddressMemcachedSessionComparator implements MemcachedSessionComparator, Serializable {
+  static final long serialVersionUID = -1L;
+
+  public int compare(Session o1, Session o2) {
+    MemcachedSession session1 = (MemcachedSession) o1;
+    MemcachedSession session2 = (MemcachedSession) o2;
+    if (session1 == null) {
+      return -1;
+    }
+    if (session2 == null) {
+      return 1;
+    }
+    return session1.getInetSocketAddressWrapper().getInetSocketAddress().toString()
```

Comment:
It may throw NPE here? I am not sure whether  `getInetSocketAddress` will return null pointer, but it prefer check it here before `toString` for robustness.

Reply:
Ok. I agree with your opinion. But do you think enough to check `getInetSocketAddress` ??

It looks like that  `getInetSocketAddressWrapper' also should be checked for null pointer. Because `MemcachedTCPSession` has the setter of `InetSocketAddressWrapper`. So if you want robustness, I will check the `getInetSocketAddressWrapper` and `getInetSocketAddress` with `Optional<>`.

Reply:
Geat if you can. Thanks.

Reply:
I add the logic to check null of `getInetSocketAddressWrapper` and `getInetSocketAddress`.